flake.nix

```diff
@@ -17,26 +18,13 @@
       config = system:
       let
         pkgs = nixpkgs.legacyPackages.${system};
-        swayhide = pkgs.rustPlatform.buildRustPackage rec {
-          pname = "swayhide";
-          version = "0.2.1";
-          src = ./.;
-          cargoSha256 = "sha256-zsWixMdh5QHzjG8OdYVXQqjjuBDhTeqX7iAFeOyEOCk=";
-          nativeBuildInputs = with pkgs; [ installShellFiles ];
-
-          postInstall = ''
-            installShellCompletion \
-              --name ${pname} completions/swayhide.bash \
-              --name ${pname}.fish completions/swayhide.fish \
-              --name _${pname} completions/swayhide.zsh
-          '';
-        };
+        swayhide = pkgs.callPackage ./. { };
       in {
         packages.${system}.default = swayhide;
 
```

Comment:
You can move the `overlays.default = import ./overlay.nix;` to here instead, and skip the attribute set merge. I find that more clean